pkgs/development/tools/ocaml/ocamlformat/default.nix

```diff
@@ -72,5 +72,9 @@ rec {
     version = "0.21.0";
   };
 
-  ocamlformat = ocamlformat_0_21_0;
+  ocamlformat_0_22_4 = mkOCamlformat {
+    version = "0.22.4";
+  };
+
+  ocamlformat = ocamlformat_0_22_4;
```

Comment:
BTW, in my opinion we can keep just the versions that Nixpkgs uses as deps + the most recent one.

If someone needs something different they always can override it (or call `mkOCamlformat`, assuming this function is exposed).

Reply:
@vbgl Is removing older versions of `ocamlformat` a good idea?

Reply:
We should maintain aliases for at least one major release, so we would probably keep it until 22.11 at least.

However, AFAIK there is no formal process to the removal of aliases. So they can be kept for a long time.

Reply:
> @vbgl Is removing older versions of `ocamlformat` a good idea?

If it is not, it would be a good idea to have at least a good justification to keep those old versions. Because maintaining all those versions is not really healthy.

Reply:
Removing some versions from `all-packages.nix` seems to be a good idea. There are certainly too many of them.

My personal preference would be to have just one `ocamlformat` attribute that can be easily overridden to select a specific version (e.g., `ocamlformat.override { version = "0.14.1"; }`). Just beware of code rot.

Reply:
> Is removing older versions of ocamlformat a good idea?
The _script_ over the comment says that the minimum version in use is `0.19.0` so I'm keeping this. However I feel that we shouldn't have so many versions declared on our repository because this can become quite a difficult task to maintain... minor updates on old versions might change the package's/dependency's derivations, aliases that stick forever...

Reply:
Instead of keeping older versions, give a try to what vbgl suggested.

> My personal preference would be to have just one ocamlformat attribute that can be easily overridden to select a specific version (e.g., ocamlformat.override { version = "0.14.1"; }). Just beware of code rot.

Reply:
I am kinda of wary of doing what @vbgl suggested in this PR. This can be done later on in another PR instead.

Reply:
Ok, then.